### PR TITLE
Vs2017 upgrade

### PR DESCRIPTION
**Pull request by optical**
This updates the project system to the updated csproj format and updates the build system to match.

I bumped a few of the MS dependencies to the official versions wherever beta/RC/preview was used - mainly in the tests. I've also attempted to preserve the xunit xml documentation which no longer works in the same way prior to the switch. https://github.com/dotnet/cli/issues/4921 outlines this (you're aware by the looks).

Travis can now run tests, although only for the netcoreapp target.

I tried to keep the csproj/cake similar to your other repositories & blog post.

### CODE REVIEW

**Comment by optical** [comment]
@RehanSaeed Any chance you could give this a look over?

File: Source/ExceptionFinderTool/ExceptionFinderTool.csproj [hunk 1]
```diff
@@ -0,0 +1,23 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+  <PropertyGroup Label="Build">
+    <TargetFrameworks>net45;netcoreapp1.0</TargetFrameworks>
+    <AssemblyName>ExceptionFinderTool</AssemblyName>
+    <CodeAnalysisRuleSet>../../MinimumRecommendedRulesWithStyleCop.ruleset</CodeAnalysisRuleSet>
+    <OutputType>Exe</OutputType>
+    <RuntimeFrameworkVersion Condition=" '$(TargetFramework)' == 'netcoreapp1.0' ">1.1.1</RuntimeFrameworkVersion>
+    <NetStandardImplicitPackageVersion Condition=" '$(TargetFramework)' == 'netcoreapp1.0' ">1.6.1</NetStandardImplicitPackageVersion>
```
Comment (RehanSaeed) [inline Source/ExceptionFinderTool/ExceptionFinderTool.csproj]: I don't think you need `NetStandardImplicitPackageVersion`.

File: Source/Serilog.Exceptions/Serilog.Exceptions.csproj [hunk 2]
```diff
@@ -0,0 +1,48 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+  <PropertyGroup Label="Build">
+    <TreatWarningsAsErrors>true</TreatWarningsAsErrors>
+    <TargetFrameworks>net45;netstandard1.3;netstandard1.6</TargetFrameworks>
+    <CodeAnalysisRuleSet>../../MinimumRecommendedRulesWithStyleCop.ruleset</CodeAnalysisRuleSet>
+    </PropertyGroup>
+
+  <PropertyGroup Label="Package">
+    <Description>Log exception details and custom properties that are not output in Exception.ToString().</Description>
+    <AssemblyTitle>Serilog Exceptions</AssemblyTitle>
```
Comment (RehanSaeed) [inline Source/Serilog.Exceptions/Serilog.Exceptions.csproj]: Only need to specify the following here (Copied from another project of mine):

```
<VersionPrefix>1.0.1</VersionPrefix>
    <Authors>Muhammad Rehan Saeed (RehanSaeed.com)</Authors>
    <Product>Boilerplate Framework</Product>
    <Description>Provides boilerplate framework code for a .NET project.</Description>
    <Copyright>Copyright © Muhammad Rehan Saeed. All rights Reserved</Copyright>
    <PackageRequireLicenseAcceptance>true</PackageRequireLicenseAcceptance>
    <PackageLicenseUrl>https://github.com/ASP-NET-MVC-Boilerplate/Framework/blob/master/LICENSE</PackageLicenseUrl>
    <PackageProjectUrl>https://github.com/ASP-NET-MVC-Boilerplate/Framework</PackageProjectUrl>
    <PackageIconUrl>https://raw.githubusercontent.com/RehanSaeed/ASP.NET-MVC-Boilerplate/master/Images/Icon.png</PackageIconUrl>
    <RepositoryUrl>https://github.com/ASP-NET-MVC-Boilerplate/Framework.git</RepositoryUrl>
    <RepositoryType>git</RepositoryType>
    <PackageTags>Boilerplate;Muhammad Rehan Saeed;Framework</PackageTags>
    <PackageReleaseNotes>Initial version.</PackageReleaseNotes>
```

I move the signing properties to a separate section (Copied from another project):

```
<PropertyGroup Label="Signing">
    <SignAssembly>true</SignAssembly>
    <AssemblyOriginatorKeyFile>../../../Key.snk</AssemblyOriginatorKeyFile>
    <PublicSign Condition=" '$(OS)' != 'Windows_NT' ">true</PublicSign>
  </PropertyGroup>
```

File: Source/Serilog.Exceptions/Serilog.Exceptions.csproj [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    <PackageId>Serilog.Exceptions</PackageId>
+    <PackageTags>Serilog;Exception;Log;Logging;Detail;Details</PackageTags>
+    <PackageReleaseNotes>Updated to target Serilog 2.4.0.</PackageReleaseNotes>
+    <PackageIconUrl>https://raw.githubusercontent.com/RehanSaeed/Serilog.Exceptions/master/Images/Serilog%20Community%20256x256.png</PackageIconUrl>
+    <PackageProjectUrl>https://github.com/RehanSaeed/Serilog.Exceptions</PackageProjectUrl>
+    <PackageLicenseUrl>https://github.com/RehanSaeed/Serilog.Exceptions/blob/master/LICENSE</PackageLicenseUrl>
+    <PackageRequireLicenseAcceptance>true</PackageRequireLicenseAcceptance>
+    <RepositoryType>git</RepositoryType>
+    <RepositoryUrl>https://github.com/RehanSaeed/Serilog.Exceptions.git</RepositoryUrl>
+    <NetStandardImplicitPackageVersion Condition=" '$(TargetFramework)' == 'netstandard1.3' ">1.6.0</NetStandardImplicitPackageVersion>
```
Comment (RehanSaeed) [inline Source/Serilog.Exceptions/Serilog.Exceptions.csproj]: Don't need `NetStandardImplicitPackageVersion`

**Comment by RehanSaeed** [comment]
Also, apologies for not coming to this sooner. If you haven't got time to fix the small issues, let me know and I'll sort it myself in the next few days.

**Comment by optical** [comment]
Hey @RehanSaeed - sorry I didn't get back to you on the fixups here. Have been vacationing for the last week so have had minimal time to follow up on this. 

Thanks for fixing it up and merging it! :)

**Comment by RehanSaeed** [comment]
Nah, we're all in the same boat in open source. We do things when we have time. Thanks for your efforts.

**Pull request merged**